addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/ClassicWiFiLEDDriver.java

```diff
@@ -137,28 +152,44 @@ public synchronized void incProgramSpeed(int step) throws IOException {
         sendLEDData(ledState);
     }
 
-    private void sendLEDData(LEDStateDTO ledState) throws IOException {
-        logger.debug("Setting LED State to {}", ledState);
+    private synchronized void sendLEDData(final LEDStateDTO ledState) throws IOException {
+        cachedLedStatus = ledState;
+        if (!ledUpdateFuture.isDone()) {
+            ledUpdateFuture.cancel(true);
+        }
 
+        final byte[] bytes;
         int program = Integer.valueOf(ledState.getProgram().toString());
         if (program == 0x61) {
             // "normal" program: set color etc.
-            byte r = (byte) (ledState.getColor().getRed() & 0xFF);
-            byte g = (byte) (ledState.getColor().getGreen() & 0xFF);
-            byte b = (byte) (ledState.getColor().getBlue() & 0xFF);
+            byte r = (byte) (ledState.getRGB()>>16 & 0xFF);
+            byte g = (byte) (ledState.getRGB()>>8 & 0xFF);
+            byte b = (byte) (ledState.getRGB() & 0xFF);
             byte w = (byte) (((int) (ledState.getWhite().doubleValue() * 255 / 100)) & 0xFF);
             byte w2 = (byte) (((int) (ledState.getWhite2().doubleValue() * 255 / 100)) & 0xFF);
 
-            byte[] bytes = getBytesForColor(r, g, b, w, w2);
+            bytes = getBytesForColor(r, g, b, w, w2);
 
-            sendRaw(bytes);
         } else {
             // program selected
             byte p = (byte) (program & 0xFF);
             byte s = (byte) (((100 - ledState.getProgramSpeed().intValue()) * 0x1F / 100) & 0xFF);
-            byte[] data = { 0x61, p, s };
-            sendRaw(data);
+            bytes = new byte[]{ 0x61, p, s };
         }
+
+        ledUpdateFuture = updateScheduler.submit(() -> {
+            try {
+                Thread.sleep(WAIT_UPDATE_LED_FOR_MS);
+                logger.debug("Setting LED State to {}", bytesToHex(bytes));
+                sendRaw(bytes);
+                return true;
+            } catch (IOException e) {
+                logger.warn("Exception occurred while sending command to LED", e);
```

Comment:
Shouldn't this rather be debug and the Thing status being set to OFFLINE?

Reply:
@kaikreuzer I do agree, but its a pretty big refactoring because the driver is not aware of the thingHandler, what would you prefer passing the handler of adding some listener interface?